src/apply.jl

```diff
@@ -118,7 +118,8 @@ the usual [`Transform`](@ref) being invoked.
 function apply_append(A::AbstractArray, t; append_dim, kwargs...)::AbstractArray
     result = apply(A, t; kwargs...)
     result = _postformat(cardinality(t), result, A, append_dim)
-    return cat(A, result; dims=append_dim)
+    # Call parent to avoid clashing axis/key names in concatenated result
+    return cat(A, parent(result); dims=append_dim)
```

Comment:
Are we sure this is OK for if the arrays are some kind of `SubArray` or `ReshapedArray` or `Symmetric` etc?

Reply:
maybe add them to tests?

Reply:
added tests for NamedDimsArrays, SubArrays, and ReshapedArrays.

`Symmetric` is a bit odd because you can't call `replace` on it so it's not compatible with the test suite.

```julia
ERROR: ArgumentError: Cannot set a non-diagonal index in a symmetric matrix
Stacktrace:
```